smack-experimental/src/main/java/org/jivesoftware/smackx/chat_markers/ChatMarkersManager.java

```diff
@@ -70,6 +109,76 @@ public static synchronized ChatMarkersManager getInstanceFor(XMPPConnection conn
 
     private ChatMarkersManager(XMPPConnection connection) {
         super(connection);
+        connection.addStanzaInterceptor(new StanzaListener() {
+            @Override
+            public void processStanza(Stanza packet)
+                    throws
+                    NotConnectedException,
+                    InterruptedException,
+                    SmackException.NotLoggedInException {
+                Message message = (Message) packet;
+                if (!isEligibleForChatMarkers(message)) {
+                    return;
+                }
+
+                if (!MessageWithBodiesFilter.INSTANCE.accept(message)) {
+                    return;
+                }
+
+                // if message already has a chatMarkerExtension, then do nothing,
+                if (ChatMarkersFilter.INSTANCE.accept(message)) {
+                    return;
+                }
+
+                // otherwise add a markable extension,
+                message.addExtension(new ChatMarkersElements.MarkableExtension());
+            }
+        }, MessageTypeFilter.NORMAL_OR_CHAT_OR_GROUPCHAT);
+
+        connection.addSyncStanzaListener(new StanzaListener() {
+            @Override
+            public void processStanza(Stanza packet)
+                    throws
+                    NotConnectedException,
+                    InterruptedException,
+                    SmackException.NotLoggedInException {
+                final Message message = (Message) packet;
+                if (!isEligibleForChatMarkers(message)) {
```

Comment:
Same here: Those two should be filters used in the filter argument of `addSyncStanzaListener()`, not in the listener code itself.

Reply:
for this I'm not sure because if you take a look at the `isEligibleForChatMarkers()` the logic there is a little bit more complex:
1. It analize if the message has a stanzaId (I couldn't find a filter for it. There are for a specific stanzaId but not for just check it)
2. messages with or without `chat state` can be accepted, but in case they have it, we are going to accept only ones with `active` state.

If there is a way to add this to a custom filter, please give me a light to do it :D